### ISSUE TITLE
Opening conversation with "Unknown" crashes Signal

### ISSUE THREAD

**Issue by pgerber** [signalapp/Signal-Android#11202]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I have one conversation with a person labeled *Unknown* that contains one message but when I try to open the conversation Signal crashes.

### Screenshots

[
![signal-2021-04-13-190950](https://user-images.githubusercontent.com/11895833/114594338-ad29af00-9c7c-11eb-8cc4-8cc77edef998.jpeg)
](url)

### Device info

**Device:** Motorola One
**Android version:** 10
**Signal version:**5.8.0

### Link to debug log

```
04-13 19:09:59.178 15987 15987 E AndroidRuntime: FATAL EXCEPTION: main
04-13 19:09:59.178 15987 15987 E AndroidRuntime: Process: org.thoughtcrime.securesms, PID: 15987
04-13 19:09:59.178 15987 15987 E AndroidRuntime: java.lang.AssertionError: No options of default type!
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at org.thoughtcrime.securesms.TransportOptions.getSelectedTransport(TransportOptions.java:108)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at org.thoughtcrime.securesms.TransportOptions.notifyTransportChangeListeners(TransportOptions.java:210)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at org.thoughtcrime.securesms.TransportOptions.setDefaultTransport(TransportOptions.java:67)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at org.thoughtcrime.securesms.components.SendButton.setDefaultTransport(SendButton.java:87)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at org.thoughtcrime.securesms.conversation.ConversationActivity.handleSecurityChange(ConversationActivity.java:1529)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at org.thoughtcrime.securesms.conversation.ConversationActivity.initializeSecurity(ConversationActivity.java:1758)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at org.thoughtcrime.securesms.conversation.ConversationActivity.onCreate(ConversationActivity.java:452)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at org.thoughtcrime.securesms.PassphraseRequiredActivity.onCreate(PassphraseRequiredActivity.java:71)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7817)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7806)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1328)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3452)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3620)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2183)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:241)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7617)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
04-13 19:09:59.178 15987 15987 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:941)
```

https://debuglogs.org/74356d4f0c0963d6c73f7371b557488ac2dc13c027e8577d9636adbd42ce316c

**Comment by hiqua** [comment]
Same here:
https://debuglogs.org/4cca9abc5f999054a2505dc776062aa9b564b4cbe6a6832bae50e2368c4bfec5

I think in my case it's a group that I created via signal-desktop-beta which had a bug that created groups that didn't propagate group names, possibly resulting in some inconsistent state.

5.8.10

**Comment by hiqua** [comment]
Archiving it also makes Signal crash at start-up, maybe I should have deleted it instead, or just left it there...

**Comment by cody-signal** [comment]
Thanks for the report, we're aware and hope to get a fix out soon. Sorry the workaround right now is painful.

**Comment by AsamK** [comment]
seems to be fixed with https://github.com/signalapp/Signal-Android/commit/add65cf592aaea7de705b9dfc71764f879b866fa :)

**Comment by hiqua** [comment]
It's still broken as of 5.13.8: https://debuglogs.org/8484ea875f19487e7ddf9798c294d3539e050c11630812b03a8d2053951d699f

**Comment by AsamK** [comment]
Indeed, looks like one spot was missed ... if `isForceSmsSelection` is true for the recipient it still crashes.

> 06-18 17:35:05.176 21838 21838 E SignalUncaughtException: java.lang.AssertionError: No options of default type!
06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.TransportOptions.getSelectedTransport(TransportOptions.java:108)
06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.TransportOptions.notifyTransportChangeListeners(TransportOptions.java:210)
06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.TransportOptions.setDefaultTransport(TransportOptions.java:67)
06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.components.SendButton.setDefaultTransport(SendButton.java:87)
06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.conversation.ConversationActivity.handleSecurityChange(ConversationActivity.java:1519)
06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.conversation.ConversationActivity.initializeSecurity(ConversationActivity.java:1748)
06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.conversation.ConversationActivity.onCreate(ConversationActivity.java:461)
06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.PassphraseRequiredActivity.onCreate(PassphraseRequiredActivity.java:71)

**Comment by AsamK** [comment]
> Indeed, looks like one spot was missed ... if `isForceSmsSelection` is true for the recipient it still crashes.
> 
> > 06-18 17:35:05.176 21838 21838 E SignalUncaughtException: java.lang.AssertionError: No options of default type!
> > 06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.TransportOptions.getSelectedTransport(TransportOptions.java:108)
> > 06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.TransportOptions.notifyTransportChangeListeners(TransportOptions.java:210)
> > 06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.TransportOptions.setDefaultTransport(TransportOptions.java:67)
> > 06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.components.SendButton.setDefaultTransport(SendButton.java:87)
> > 06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.conversation.ConversationActivity.handleSecurityChange(ConversationActivity.java:1519)
> > 06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.conversation.ConversationActivity.initializeSecurity(ConversationActivity.java:1748)
> > 06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.conversation.ConversationActivity.onCreate(ConversationActivity.java:461)
> > 06-18 17:35:05.176 21838 21838 E SignalUncaughtException: 	at org.thoughtcrime.securesms.PassphraseRequiredActivity.onCreate(PassphraseRequiredActivity.java:71)

(@cody-signal did you see this?)

**Comment by greyson-signal** [comment]
There's a few layers of problems here. The core issue that is causing these 'unknown' recipients will be fixed in 5.15.3, and there will be another fix in 5.16 that should resolve this specific crash.

**Comment by hiqua** [comment]
Awesome, thanks a lot!
